add possibility to link to a station using ?ge_id=

johan12345 opened the pull request:
With this change, it is possible to directly show a station with a given ID in the GoingElectric directory. Example: http://localhost:9000/?ge_id=1234

Background: I'm developing a new open source Android app for the GoingElectric charging station directory (https://github.com/johan12345/EVMap) and would like to provide links to Chargeprice.app to directly find the cheapest price for that station. In the future, direct integration through the API would be nice as well (if that is possible free of charge), but a link to the website is probably the easiest solution for now.

johan12345 committed:
add possibility to link to a station using ?ge_id=

hoenic07 commented:
Thanks for this PR! I have this on my list for quite some time already!
To allow reusing the same query parameters for different POI/station sources (e.g. our own Chargeprice POI data), I'd like to use a more generic form: `?poi_id=1234&poi_source=going_electric`.
Also it would be great if the query parameter gets changed when clicking on another station.
Apart from that I'm fine with the PR!

johan12345 committed:
replace ge_id with poi_id and poi_source

johan12345 committed:
update poi_id and poi_source parameters when new station is selected

johan12345 committed:
add circle around selected station when using poi_id and poi_source parameters

johan12345 commented:
@hoenic07 Great, thanks! I added some changes now to replace ge_id with poi_id and poi_source and to update these parameters in the URL when a new station is selected on the map.

Also, I noticed that the circle around the selected station was not automatically shown when the app was started using these parameters, so I fixed that as well.

hoenic07 commented on src/app.js:
```diff
@@ -94,6 +101,14 @@ class App {
+      id: poiId,
+      dataAdapter: poiSource,
+      charge_points: []
+    }, ">3.7", true)
```
This doesn't yet work for POIs of other data sources (see any station in France). Also the ">3.7" (which is used for tracking), should only be set if known.
I will fix all of this (because it also requires a backend change) and try to merge it today or in the next days!

hoenic07 commented:
@johan12345 It's on production now! Great Job! Thank you!

johan12345 commented:
Thank you very much! I'll start adding the links from the [EVMap](https://github.com/johan12345/EVMap) app soon, then. :)
